Source/Widgets/Geocoder/GeocoderViewModel.js

```diff
@@ -331,14 +341,79 @@ define([
     }
 
     function updateCamera(viewModel, destination) {
-        viewModel._scene.camera.flyTo({
-            destination : destination,
-            complete: function() {
-                viewModel._complete.raiseEvent();
-            },
-            duration : viewModel._flightDuration,
-            endTransform : Matrix4.IDENTITY
-        });
+        var scene = viewModel._scene;
+        var globe = scene.globe;
+        var ellipsoid;
+        if (defined(globe)) {
+            ellipsoid = globe.ellipsoid;
+        } else {
+            ellipsoid = scene.mapProjection.ellipsoid;
+        }
+
+        var camera = scene.camera;
+        var terrainProvider = scene.terrainProvider;
+        var availability = defined(terrainProvider) ? terrainProvider.availability : undefined;
+
+        // Always expand single points so we don't zoom
+        // directly into the ground, even when not using terrain.
+        if (destination instanceof Cartesian3) {
+            var carto = ellipsoid.cartesianToCartographic(destination);
+            var offset = CesiumMath.toRadians(0.001);
+            destination = new Rectangle(
+                carto.longitude - offset,
+                carto.latitude - offset,
+                carto.longitude + offset,
+                carto.latitude + offset);
+        }
+
+        var newDestination = destination;
+        if (defined(availability)) {
+            var cartographics;
+
+            cartographics = [
```

Comment:
@mramato These don't need to be on two separate lines but oh well.  And shouldn't you use result parameters throughout this?  I know it won't really impact performance but it's still a good habit

Reply:
We use result parameters as a necessary evil for performance in hot areas of the code. Not using them makes the code cleaner and is perfectly OK if performance is not going to be an issue.